src/crs_linter/rules/variables_usage.py

```diff
@@ -0,0 +1,150 @@
+                    if "act_arg_val" in a and a["act_arg_val"] is not None:
+                        val_act_arg = re.findall(
+                            r"%\{(tx.[^%]*)}", a["act_arg_val"], re.I
+                        )
+                    for v in val_act + val_act_arg:
+                        v = v.lower().replace("tx.", "")
+                        if not re.match(r"^\d$", v, re.I):
+                            if (
+                                v not in globtxvars
+                                or phase < globtxvars[v]["phase"]
+                            ):
+                                yield LintProblem(
+                                    line=a["lineno"],
+                                    end_line=a["lineno"],
+                                    desc=f"TX variable '{v}' not set / later set (rvar) in rule {ruleid}",
+                                    rule="variables_usage",
+                                )
+                            else:
+                                globtxvars[v]["used"] = True
+                        else:
+                            if v in globtxvars:
+                                globtxvars[v]["used"] = True
+
+                if "operator_argument" in d:
+                    oparg = re.findall(r"%\{(tx.[^%]*)}", d["operator_argument"], re.I)
+                    if oparg:
+                        for o in oparg:
+                            o = o.lower()
+                            o = re.sub(r"tx\.", "", o, re.I)
+                            if (
+                                (
+                                    o not in globtxvars
+                                    or phase < globtxvars[o]["phase"]
+                                )
+                                and not re.match(r"^\d$", o)
+                                and not re.match(r"/.*/", o)
+                                and check_exists is None
+                            ):
+                                yield LintProblem(
+                                    line=d["lineno"],
+                                    end_line=d["lineno"],
+                                    desc=f"TX variable '{o}' not set / later set (OPARG) in rule {ruleid}",
+                                    rule="variables_usage",
+                                )
+                            elif (
+                                o in globtxvars
+                                and phase >= globtxvars[o]["phase"]
+                                and not re.match(r"^\d$", o)
+                                and not re.match(r"/.*/", o)
+                            ):
+                                globtxvars[o]["used"] = True
+                if "variables" in d:
+                    for v in d["variables"]:
+                        if v["variable"].lower() == "tx":
+                            if not v["counter"]:
+                                rvar = v["variable_part"].lower()
```

Comment:
And here too from [here](https://github.com/coreruleset/crs-linter/blob/444eaf7437c7f4fa9c2f2d4751b08f61f18e202a/src/crs_linter/linter.py#L501-L506).